test/models/site_test.rb

```diff
@@ -13,22 +13,58 @@ class SiteTest < ActiveSupport::TestCase
   sub_test_case "relation" do
     setup do
       @site = create(:site)
-      @post = create(:post, :whatever, site: @site)
+      @post = create(:post, :whatever, :with_credit, site: @site)
       Tempfile.open do |file|
         @site.images.create!(image: file)
       end
+      create(:serial, site: @site)
+
+      create(:fixed_page, site: @site)
+      create(:link,       site: @site)
+      create(:membership, site: @site)
+      @site.redirect_rules.create!(request_path: "/foo", destination: "/bar")
```

Comment:
So then let's use factory girl instead of `@site.redirect_rules.create!` for consistency and `reload` the site.